db/data/20220803164723_fix_end_academic_cycle_id_for_awarded_withdrawn_trainees.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class FixEndAcademicCycleIdForAwardedWithdrawnTrainees < ActiveRecord::Migration[6.1]
+  def up
+    hesa_trainees = Trainee.imported_from_hesa.where("hesa_updated_at > ?", "01/06/2022".to_date)
+
+    awarded_trainees = Trainee.where.not(id: hesa_trainees).awarded.where("updated_at > ?", "01/06/2022".to_date)
+
+    awarded_trainees.find_each do |trainee|
```

Comment:
wondering if I should add the two trainee collections together and do the loop once?

Reply:
I think it could be better as you suggested (loop over one collection).

Also, is the collection going to be around ~430 records? If so, that number is OK, but I have seen larger data migrations timeout in CI.

Reply:
Hi Mark, I've fixed this up to be one loop now. I've run the file locally against a prod backup and it seems to only take seconds to execute. The amount of records though is approx 11200. I wanted to also test this out on production data to see how it runs.

Do you think it would be better to do this data migration as a rake task instead?

Reply:
I would have a strong preference for running this as a rake task. It will give us more control of exactly when it is run and we will not risk impacting the deployment pipelines.

Reply:
Great, I'll move this to a rake task instead :)